### PR TITLE
gwenhywfar: enable Qt5, cpp GUIs

### PR DESCRIPTION
**Pull request by wrobelda**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This adds support for Qt and C++ backends, as well some additional tests. Qt5 support was fixed upstream, hence the patch and an obligatory `autoreconf`. Both can be removed once the new version is released.

### CODE REVIEW

File: Formula/gwenhywfar.rb
Comment (carlocab) [inline Formula/gwenhywfar.rb:30]: What is this patch for, and when can it be removed? Please document this in a comment.

File: Formula/gwenhywfar.rb
Comment (carlocab) [inline Formula/gwenhywfar.rb:78]: If you want to use `cmake` in the `test` block you need
```ruby
depends_on "cmake" => :test
```

File: Formula/gwenhywfar.rb
Comment (carlocab) [inline Formula/gwenhywfar.rb:46]: We should keep the C test, in addition to the C++ one.

**Comment by fxcoudert** [comment]
Why does this need several GUIs? Isn't the Cocoa GUI fully functional? We prefer native GUIs when available.

**Comment by wrobelda** [comment]
> Why does this need several GUIs? Isn't the Cocoa GUI fully functional? We prefer native GUIs when available.

Gwenhywfar isn't a standalone software; it is used by 3rd party banking apps like GnuCash and KMyMoney. I am one of the maintainers of the latter and it uses Qt5 itself, hence the reliance on Gwenhywfar's Qt5 backend.

**Comment by jonchang** [comment]
I think this is fine. It doesn't look like it's actually building a GUI, just providing backend bindings so other apps can interact using features from e.g. Qt (if I'm reading the tea leaves in Debian correctly)

**Comment by wrobelda** [comment]
> I think this is fine. It doesn't look like it's actually building a GUI, just providing backend bindings so other apps can interact using features from e.g. Qt (if I'm reading the tea leaves in Debian correctly)

It's bo

**Comment by wrobelda** [comment]
My bad, closed it accidentally.

**Review by SMillerDev — DISMISSED** [review]

**Comment by carlocab** [comment]
We should wait a bit for #70826 before merging this.

**Comment by wrobelda** [comment]
@carlocab may I ask what's your reasoning behind the wait? It will probably be a year before this and other projects (like KMyMoney) are migrated over to Qt6. Many of the existing, crucial modules won't be available until about 6.2 version – see "Migration of additionally required modules to Qt 6" here: https://www.qt.io/product/qt6/technical-specifications

**Comment by carlocab** [comment]
>It will be at least a year before this and other projects (like KMyMoney) are migrated over to Qt6.

This is exactly the problem. When that PR is merged, `depends_on "qt"` will refer to qt6. Hence, adding a qt dependency to this formula means that it actually needs to say `depends_on "qt@5"`. Except we can't do that right now, because we can't write `depends_on "qt@5"` while there is no `qt@6`.

**Comment by wrobelda** [comment]
So If it gets merged now, it will be subject to some mass renaming of "qt" to "qt@5" in all current formulas, which will inevitably happen once qt gets upgraded to qt6. I don't see what the big deal is.

**Comment by carlocab** [comment]
>it will later on be subject to some mass renaming of "qt" to "qt@5" in all current formulas

The mass renaming is happening already in that PR. Including this formula in that PR requires restarting the CI run there one more time, which I'd like to avoid doing since it already takes ages to run.

We can merge this if it looks like CI will need to be restarted in the qt PR another time (likely, but I hope not). That said, GitHub actions aren't really working at the moment, so it's not like we can merge this right now anyway.

Is this change really that urgent?

**Comment by wrobelda** [comment]
> The mass renaming is happening already in that PR.

Oh, OK, in that case it makes sense to wait. I'll update this PR once the other one is merged.

**Comment by carlocab** [comment]
The current run at #70826 is looking promising, so this PR may not need to wait very long. However, if the CI run there fails again, we can probably just merge this already (GitHub Actions API willing).

**Comment by carlocab** [comment]
Rebased to incorporate version bump in #72350, plus to get rid of the merge commit (@BrewTestBot does not like merge commits).

**Comment by wrobelda** [comment]
@carlocab what is pending to get this merged?

**Review by carlocab — APPROVED** [review]
Whoops; this got buried in other PRs. Sorry!

**Pull request closed without merging**
